### PR TITLE
Info entity context CR

### LINKED ISSUE #329: Launch in Context info entity - As a <Developer> , I want that <the "Launch in Context info entity" will be defined> so that <I can know how the context should affect the application layout>
Define the "Lunch in Context info entity" Context 
Filtering params (Facets - Sensors), Display mode (num of screens, ...),Radius from geo coordinate, ..(refinement of Definition of Context params by Idan)
The context should define the initial setting of the map when being loaded (zoom, area, rotation, ...) from context
@idanbarak @carmelorr -  
### Should be defined in the first two days of the iteration

### LINKED ISSUE #465: As a <user>, I want to <able to launch the application in "Launch in Context of Info Entity" context > so that <I can analyze the area according to the info entity>
Allow launching the app by providing a URL with the wanted context and AOI as URL params.

- [ ] update the case state according to the definition of this context by @idanbarak @carmelorr - #329 
- [ ] write all the necessary action and events to notify the app about the launched context.
- [ ] **As part of the implementation provides a readme file that describe the different params supported in the URL**

### LINKED ISSUE #466: As a <User> I want to <that the prev and next (time) overlay are presented on the imagery > so that <I can analyze the area according to the info entity>

### LINKED ISSUE #484: Add functionality to query IDAHO with max count before and after a given time

### LINKED ISSUE #485: Add a layer provider functionality

### LINKED ISSUE #488: add a context with entity from a layer
the layer should from a layer source that was pre-defined and the URL should contain the layer type and ID of some sort

### PR DESCRIPTION
**Pull request by MarkShleifer**
Added slider filter and info entity from url. The info entity is displayed on two maps and in the timeline.

closes #466, closes #484, closes #484, closes #488, closes #329, closes #485, closes #465

### CODE REVIEW

File: src/app/packages/map-facade/reducers/map.reducer.ts [hunk 1]
```diff
@@ -72,6 +76,23 @@ export function MapReducer(state: IMapState = initialMapState, action: MapAction
...
+			return { ...state, pendingMapsCount: newCount };
+
+		case MapActionTypes.SET_PENDING_OVERLAYS:
+			return { ...state, pendingOverlays: action.payload };
+
+		case MapActionTypes.REMOVE_PENDING_OVERLAY:
+			const pendingOverlaysClone = state.pendingOverlays.slice();
+			pendingOverlaysClone.splice(pendingOverlaysClone.indexOf(action.payload), 1);
+
+			return { ...state, pendingOverlays: pendingOverlaysClone };
```
Comment (yairtawil) [inline src/app/packages/map-facade/reducers/map.reducer.ts]: very good

File: src/app/packages/menu-items/filters/components/slider-filter-container/slider-filter-container.component.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    </div>
+    <div class="range">
+        {{rangeValues[0] / factor}} - {{rangeValues[1] / factor}}
+    </div>
+    <div class="max">
+        {{metadata.max}}            
+    </div>
+</div>
+
+<p-slider class='slider' [(ngModel)]="rangeValues" [range]="true" (onSlideEnd)="handleChange($event)"
```
Comment (yairtawil) [inline src/app/packages/menu-items/filters/components/slider-filter-container/slider-filter-container.component.html]: check z-index and range

File: src/app/packages/menu-items/filters/index.ts [hunk 3]
```diff
@@ -5,5 +5,6 @@ export { IFiltersState, FiltersReducer } from './reducer/filters.reducer';
 export * from './actions/filters.actions';
```
Comment (yairtawil) [inline src/app/packages/menu-items/filters/index.ts]: use *

File: src/app/packages/menu-items/filters/models/metadata/enum-filter-metadata.ts [hunk 4]
```diff
@@ -65,7 +99,7 @@ export class EnumFilterMetadata implements FilterMetadata {
 	}
 
 	isFiltered(): boolean {
-		return Array.from(this.enumsFields.values()).some((value: { count: number, isChecked: boolean }) => !value.isChecked);
+		return Array.from(this.enumsFields.values()).some((value: { count: number, isChecked: boolean }) => value.isChecked);
```
Comment (yairtawil) [inline src/app/packages/menu-items/filters/models/metadata/enum-filter-metadata.ts]: check

File: src/app/packages/menu-items/filters/models/metadata/filter-metadata.interface.ts [hunk 5]
```diff
@@ -1,12 +1,12 @@
 export abstract class FilterMetadata {
 	type: string;
 
-	enumsFields: Map<string, { count: number, isChecked: boolean }>;
```
Comment (yairtawil) [inline src/app/packages/menu-items/filters/models/metadata/filter-metadata.interface.ts]: naha

File: src/app/packages/overlays/actions/overlays.actions.ts [hunk 6]
```diff
@@ -93,6 +94,13 @@ export class DisplayOverlayFromStoreAction implements Action {
 	}
 }
 
+export class DisplayMultipleOverlaysFromStoreAction implements Action {
+	type = OverlaysActionTypes.DISPLAY_MULTIPLE_OVERLAYS_FROM_STORE;
+
+	constructor(public payload: string[]) {
```
Comment (yairtawil) [inline src/app/packages/overlays/actions/overlays.actions.ts]: max 4 overlay typescript check

File: .angular-cli.json [hunk 7]
```diff
@@ -21,7 +21,10 @@
 			"showCircularDependencies": false,
 			"prefix": "ansyn",
 			"styles": [
-				"./styles/styles.less"
+				"./styles/styles.less",
```
[the remark this answers is not in the document]
Comment (MarkShleifer) [reply .angular-cli.json]: 👍

File: confd/production.tmpl [hunk 8]
```diff
@@ -113,7 +113,8 @@
     "filtersConfig": {
         "filters": [
             { "modelName": "sensorType", "displayName": "Sensor Type", "type": "Enum" },
-            { "modelName": "sensorName", "displayName": "Sensor Name", "type": "Enum" }
+            { "modelName": "sensorName", "displayName": "Sensor Name", "type": "Enum" },
+            { "modelName": "bestResolution", "displayName": "Resolution", "type": "Slider" }
```
[the remark this answers is not in the document]
Comment (MarkShleifer) [reply confd/production.tmpl]: 👍

File: src/app/app/app-effects/effects/context/context-entity.app.effect.ts [hunk 9]
```diff
@@ -6,35 +6,81 @@ import { IAppState } from '../../app.effects.module';
...
+					shape: 'star'
+				} as OverlaySpecialObject;
+				actions.push(new SetSpecialObjectsActionStore([specialObject]));
+			})
+
+			return actions;
+		});
+
+	/**
+ * @type Effect
```
[the remark this answers is not in the document]
Comment (MarkShleifer) [reply src/app/app/app-effects/effects/context/context-entity.app.effect.ts]: 👍

File: src/app/packages/menu-items/filters/models/metadata/filter-metadata.interface.ts (hunk 5 above)
Comment (MarkShleifer) [reply src/app/packages/menu-items/filters/models/metadata/filter-metadata.interface.ts]: 👍

File: src/app/packages/map-facade/reducers/map.reducer.ts (hunk 1 above)
Comment (MarkShleifer) [reply src/app/packages/map-facade/reducers/map.reducer.ts]: :+1

File: src/app/app/app-effects/effects/overlays.app.effects.ts [hunk 10]
```diff
@@ -172,11 +228,126 @@ export class OverlaysAppEffects {
...
+	displayTwoNearestOverlay$: Observable<any> = this.actions$
+		.ofType(OverlaysActionTypes.SET_FILTERS)
+		.filter(action => this.casesService.contextValues.defaultOverlay === 'nearest')
+		.withLatestFrom(this.store$.select(overlaysStateSelector), (action, overlays: IOverlaysState) => {
+			return [overlays.filteredOverlays, overlays.overlays];
+		})
+		.filter(([filteredOverlays, overlays]: [string[], Map<string, Overlay>]) => !isEmpty(filteredOverlays))
+		.map(([filteredOverlays, overlays]: [string[], Map<string, Overlay>]) => {
+
+			const overlaysBefore = filter(filteredOverlays, function (overlay) {
```
[the remark this answers is not in the document]
Comment (MarkShleifer) [reply src/app/app/app-effects/effects/overlays.app.effects.ts]: 👍

File: src/app/app/app-effects/effects/overlays.app.effects.ts [hunk 11]
```diff
@@ -172,11 +228,126 @@ export class OverlaysAppEffects {
...
+				return overlays.get(overlay).photoTime > this.casesService.contextValues.time;
+			}.bind(this))
+				.sort(function (a, b) {
+					return new Date(overlays.get(a).photoTime).getTime() - 
+					new Date(overlays.get(b).photoTime).getTime();
+				});
+
+			this.casesService.contextValues.defaultOverlay = '';
+
+			return new DisplayMultipleOverlaysFromStoreAction([first(overlaysBefore), first(overlaysAfter)]);
```
[the remark this answers is not in the document]
Comment (MarkShleifer) [reply src/app/app/app-effects/effects/overlays.app.effects.ts]: 👍

File: src/app/packages/map-facade/actions/map.actions.ts [hunk 12]
```diff
@@ -278,3 +289,30 @@ export class SetFavoriteAction implements Action {
...
+	}
+}
+
+export class DecreasePendingMapsCountAction implements Action {
+	type: string = MapActionTypes.DECREASE_PENDING_MAPS_COUNT;
+
+	constructor() { }
+}
+
+export class SetPendingOverlaysAction implements Action {
```
[the remark this answers is not in the document]
Comment (MarkShleifer) [reply src/app/packages/map-facade/actions/map.actions.ts]: :+1

File: src/app/packages/menu-items/filters/components/slider-filter-container/slider-filter-container.component.html (hunk 2 above)
Comment (MarkShleifer) [reply src/app/packages/menu-items/filters/components/slider-filter-container/slider-filter-container.component.html]: 👍

File: src/app/packages/menu-items/filters/index.ts (hunk 3 above)
Comment (MarkShleifer) [reply src/app/packages/menu-items/filters/index.ts]: 👍

File: src/app/packages/menu-items/filters/models/metadata/enum-filter-metadata.ts (hunk 4 above)
Comment (MarkShleifer) [reply src/app/packages/menu-items/filters/models/metadata/enum-filter-metadata.ts]: 👍

File: src/app/packages/overlays/actions/overlays.actions.ts (hunk 6 above)
Comment (MarkShleifer) [reply src/app/packages/overlays/actions/overlays.actions.ts]: 👍

**Pull request merged**